src/test/operators/join_index_test.cpp

```diff
@@ -121,345 +114,119 @@ class JoinIndexTest : public BaseTest {
-                         {{ColumnID{0}, ColumnID{0}}, PredicateCondition::GreaterThanEquals}, JoinMode::FullOuter,
-                         "resources/test_data/tbl/join_operators/int_greaterequal_outer_join.tbl", 1);
-}
-
-TYPED_TEST(JoinIndexTest, GreaterEqualInnerJoin2) {
-  // Joining two Integer Columns
-  this->test_join_output(this->_table_wrapper_i, this->_table_wrapper_j,
-                         {{ColumnID{0}, ColumnID{0}}, PredicateCondition::GreaterThanEquals}, JoinMode::Inner,
-                         "resources/test_data/tbl/join_operators/int_greaterequal_inner_join_2.tbl", 1);
-}
-
-TYPED_TEST(JoinIndexTest, NotEqualInnerJoin) {
-  // Joining two Integer Columns
-  this->test_join_output(this->_table_wrapper_a, this->_table_wrapper_b,
-                         {{ColumnID{0}, ColumnID{0}}, PredicateCondition::NotEquals}, JoinMode::Inner,
-                         "resources/test_data/tbl/join_operators/int_notequal_inner_join.tbl", 1);
-  // Joining two Float Columns
-  this->test_join_output(this->_table_wrapper_a, this->_table_wrapper_b,
-                         {{ColumnID{1}, ColumnID{1}}, PredicateCondition::NotEquals}, JoinMode::Inner,
-                         "resources/test_data/tbl/join_operators/float_notequal_inner_join.tbl", 1);
-}
-
-TYPED_TEST(JoinIndexTest, NotEqualInnerJoinDict) {
-  // Joining two Integer Columns
-  this->test_join_output(this->_table_wrapper_a, this->_table_wrapper_b,
-                         {{ColumnID{0}, ColumnID{0}}, PredicateCondition::NotEquals}, JoinMode::Inner,
-                         "resources/test_data/tbl/join_operators/int_notequal_inner_join.tbl", 1);
-  // Joining two Float Columns
-  this->test_join_output(this->_table_wrapper_a, this->_table_wrapper_b,
-                         {{ColumnID{1}, ColumnID{1}}, PredicateCondition::NotEquals}, JoinMode::Inner,
-                         "resources/test_data/tbl/join_operators/float_notequal_inner_join.tbl", 1);
-}
+  performance_data.executed = true;
+  performance_data.has_output = true;
+  performance_data.output_row_count = 2u;
+  performance_data.output_chunk_count = 1u;
+  performance_data.walltime = std::chrono::nanoseconds{999u};
+  performance_data.chunks_scanned_with_index = 10u;
+  performance_data.chunks_scanned_without_index = 5u;
+  stream << performance_data;
+  EXPECT_EQ(stream.str(), "2 row(s) in 1 chunk(s), 999 ns / 10 of 15 chunks used an index");
 
-TYPED_TEST(JoinIndexTest, JoinLessThanOnDictAndDict) {
-  this->test_join_output(this->_table_wrapper_a, this->_table_wrapper_b,
-                         {{ColumnID{0}, ColumnID{0}}, PredicateCondition::LessThanEquals}, JoinMode::Inner,
-                         "resources/test_data/tbl/join_operators/int_float_leq_dict.tbl", 1);
+  stream.str("");
+  performance_data.output_to_stream(stream, DescriptionMode::MultiLine);
+  EXPECT_EQ(stream.str(), "2 row(s) in 1 chunk(s), 999 ns\n10 of 15 chunks used an index");
```

Comment:
(Really minor:) Not a fan of reusing variable names. You could add a scope for the two tests.